### PR TITLE
feat: improve discardcloudservice filter in yurthub component (#1924)

### LINKED ISSUE #1924: [feature request]improve discardcloudservice filter in yurthub component
**What would you like to be added**:
Because clients on edges nodes can not connect cloud applications through pod ip, so LoadBalancer service for cloud application is discarded in yurthub component in order to make kube-proxy unaware of LoadBalancer service on edge nodes. and only LoadBalancer service with annotation `openyurt.io/skip-discard=true` will be skipped. 
the following are detail codes:

https://github.com/openyurtio/openyurt/blob/81d21b7f438be4fb912ebe9748f0c9a4a9296f5f/pkg/yurthub/filter/discardcloudservice/filter.go#L81-L86

but in the proposal: https://github.com/openyurtio/openyurt/blob/master/docs/proposals/20231121-high-availability-of-edge-services.md, LoadBalancer service across NodePools will be supported, this means that LoadBalancer service should be aware of on edge nodes.

so we will improve discardcloudservice filter in yurthub component, and only discard LoadBalancer service which has annotation `svc.openyurt.io/discard=true`, this means that LoadBalancer service will not be discarded by default.

**others**
/kind feature

### PR DESCRIPTION
**Pull request by huangchenzhao**
#### What this PR does / why we need it:
have improved discardcloudservice filter in yurthub component

only discarding LoadBalancer service which has annotation `svc.openyurt.io/discard=true`, instead of which doesn't have `openyurt.io/skip-discard=true`.
#### Which issue(s) this PR fixes:

Fixes #1924

### CODE REVIEW

**Comment by huangchenzhao** [comment]
It seems unit tests in 'discardcloudservice filter' should be modified to ensure synchronization with the changed code, I will fix it soon.

File: pkg/yurthub/filter/constant.go [hunk 1]
```diff
@@ -46,6 +46,11 @@ const (
 	// If end users want to use specified LB service at the edge side,
 	// End users should add annotation["openyurt.io/skip-discard"]="true" for LB service.
 	SkipDiscardServiceAnnotation = "openyurt.io/skip-discard"
+
+	// DiscardServiceAnnotation is annotation used by LB service.
+	// If end users want to discard specified LB service at the edge side,
+	// End users should add annotation["openyurt.io/discard"]="true" for LB service.
+	DiscardServiceAnnotation = "openyurt.io/discard"
```
Comment (rambohe-ch) [inline pkg/yurthub/filter/constant.go:53]: openyurt.io/discard --> svc.openyurt.io/discard

**Comment by rambohe-ch** [comment]
> It seems unit tests in 'discardcloudservice filter' should be modified to ensure synchronization with the changed code, I will fix it soon.

yes, unit tests also should be tuned.

File: pkg/yurthub/filter/constant.go (hunk 1 above)
Comment (huangchenzhao) [reply pkg/yurthub/filter/constant.go]: That's right, fixed.

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**